### PR TITLE
update list of released versions

### PR DESCRIPTION
**Pull request by RichieEscarez**
- Add release-0.4 to repo list of available doc versions
- Add link to knative.dev
- Suggest site for users and repo for contributors

### CODE REVIEW

File: doc-releases.md [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+## Documentation website
+
+Use the Knative Documentation website to learn about the product:
+
+- [`https://www.knative.dev`](https://www.knative.dev)
+
+
+## `knative/docs` branches
+
+If want to fix or add content to a past release, you can find the source files
```
Comment (samodell) [inline doc-releases.md]:
```suggestion
If you want to fix or add content to a past release, you can find the source files
```

File: doc-releases.md [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
+in the following branches:
```
Comment (samodell) [inline doc-releases.md]: This feels a little strange. Should this heading be combined with the "released versions" section below, or did you mean to add more here?

Comment (RichieEscarez) [reply doc-releases.md]: there are released branches and a development branch but really just want to be clear that you should use the website

Comment (RichieEscarez) [reply doc-releases.md]: converted it to a list, ptal

**Pull request merged**